gitlab/utils.py

```diff
@@ -191,8 +191,7 @@ def warn(
     stacklevel = 1
     warning_from = ""
     for stacklevel, frame in enumerate(reversed(stack), start=1):
-        if stacklevel == 2:
-            warning_from = f" (python-gitlab: {frame.filename}:{frame.lineno})"
+        warning_from = f" (python-gitlab: {frame.filename}:{frame.lineno})"
```

Comment:
Should we also adapt the message passed in here from https://github.com/python-gitlab/python-gitlab/blob/1e37b8fb1109fbfe01b908c4663971876e31c18c/gitlab/client.py#L957-L962?

Since this happens in `http_list` and when called form custom methods like `refs()`, the `list()` method is never involved so I'm wondering if we need something a bit more generic to avoid the confusion reported.

But I would also still need to see what OP's error looked like, maybe there is already a hint as well.

Reply:
@nejch That is a separate issue I think. As that is from the caller of this method. I'm just trying to fix this so the correct user's calling filename appears.

Agree that we should figure out if we want a better message.